lib/tests/tap_syntax.rb

```diff
@@ -14,6 +14,19 @@ def run!(args:)
 
         test "brew", "readall", "--aliases", tap.name
         test "brew", "audit", "--tap=#{tap.name}"
+
+        return if ["push", "merge_group"].exclude?(ENV["GITHUB_EVENT_NAME"])
+
+        FileUtils.mkdir_p "api"
+        Pathname("api").cd do
+          if tap.core_tap?
+            test "brew", "generate-formula-api"
+          elsif tap.name.start_with?("homebrew/cask")
+            test "brew", "generate-cask-api"
+          end
+        end
+      ensure
+        FileUtils.rm_rf "api"
```

Comment:
This seems very dangerous for non-GitHub Actions (or non-local).

Reply:
If you want it to be a subdirectory: at least move this to a begin block (or separate method) after the conditional above - so that the return from the conditional won't trigger the ensure block.

Reply:
"Very dangerous" seems a little strong given how widely `brew test-bot` is used but: sure, I'll open a PR.

Reply:
https://github.com/Homebrew/homebrew-test-bot/pull/921

Reply:
We've had contributors run `test-bot` commands on their own machines before.

Reply:
We have and deleting a directory called `api` does not seem on the grounds of "very dangerous" given what else can be done with flags here.

Reply:
People don't go passing `--cleanup` randomly (it doesn't even appear in CI logs because we do it invisibly). The default behaviour of test-bot without that flag is safe to run.